### PR TITLE
Changes to create and delete project and other bug fixes

### PR DESCRIPTION
**Pull request by akabhask**
Changes to create and delete project and other bug fixes

### CODE REVIEW

File: plugins/modules/template_intent.py [hunk 1]
```diff
@@ -896,163 +1037,259 @@ def requires_update(self):
...
+            name = f"templateName: {template_params.get('templateName')}"
 
-            self.result['response'] = task_details if task_details else response
+        response = self.dnac._exec(
+            family="configuration_templates",
+            function=deletion_value,
+            params=params_key,
+        )
 
+        task_id = response.get("response").get("taskId")
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:1199]: if task_id is not true, then we can return with failed message.

File: plugins/modules/template_intent.py [hunk 2]
```diff
@@ -896,163 +1037,259 @@ def requires_update(self):
...
+        return self
+
+    def get_diff_delete(self, template_config):
+        """Delete projects or templates in DNAC with fields provided in playbook."""
+
+        is_template_found = self.have_create_template.get("template_found")
+        is_project_found = self.have_create_project.get("project_found")
+        template_params = self.want_create.get("template_params")
+
+        if is_project_found:
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:1224]: if project is not found, then we can return from here.

**Review by madhansansel — COMMENTED** [review]
Minor changes, we will discuss before commit the changes.

File: plugins/modules/template_intent.py [hunk 3]
```diff
@@ -277,7 +277,7 @@
         description: Description of template.
         type: str
       deviceTypes:
-        description: Configuration Template Create's deviceTypes.
+        description: Configuration Template Create's deviceTypes. This field is mandatory to create a new template.
```
Comment (madhansansel) [inline plugins/modules/template_intent.py:280]: Can we change the description of devicetypes, rollbackTemplateParams and others

**Review by madhansansel — APPROVED** [review]

File: plugins/modules/template_intent.py [hunk 4]
```diff
@@ -649,219 +649,360 @@
...
+
+        # DNAC returns project details even if the substring matches.
+        # Hence check the projectName retrieved from DNAC.
+        if project_details:
+            fetched_project_name = project_details[0].get('name')
+            if fetched_project_name == given_project_name:
+                have_create_project["project_found"] = True
+                template_available = project_details[0].get('templates')
+                have_create_project["id"] = project_details[0].get("id")
+                have_create_project["isDeletable"] = project_details[0].get("isDeletable")
```
[the remark this answers is not in the document]
Comment (akabhask) [reply plugins/modules/template_intent.py:834]: Fixed.

File: plugins/modules/template_intent.py [hunk 5]
```diff
@@ -649,219 +649,360 @@
...
+            function=creation_value,
+            op_modifies=True,
+            params=params_key,
+        )
+        if not isinstance(response, dict):
+            return creation_id, created
+
+        task_id = response.get("response").get("taskId")
+        if not task_id:
+            return creation_id, created
```
[the remark this answers is not in the document]
Comment (akabhask) [reply plugins/modules/template_intent.py:966]: fixed.

File: plugins/modules/template_intent.py (hunk 1 above)
Comment (akabhask) [reply plugins/modules/template_intent.py:1199]: fixed.

File: plugins/modules/template_intent.py (hunk 3 above)
Comment (akabhask) [reply plugins/modules/template_intent.py:280]: Fixed.

File: plugins/modules/template_intent.py [hunk 6]
```diff
@@ -649,219 +649,360 @@
...
+
+        self.log(f"New {name} created with id {creation_id}")
+        return creation_id, created
 
     def requires_update(self):
-        current_obj = self.have_create.get("template")
+        """Check if the template config given requires update."""
+
+        if self.have_create_template.get("isCommitPending"):
+            return True
```
[the remark this answers is not in the document]
Comment (akabhask) [reply plugins/modules/template_intent.py:1003]: fixed.

**Pull request merged**